analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDBExtension.java

```diff
@@ -166,6 +174,14 @@ public void consume(String record) {
                     }
                 });
             } catch (Exception expected) {
+                if (expected instanceof DataAccessException) {
+                    // The error codes starting with 57P0 are related to the DB connection issues.
+                    // See https://www.postgresql.org/docs/current/errcodes-appendix.html
+                    if (((DataAccessException) expected).sqlState().contains("57P0")) {
```

Comment:
Why do we need this weird state check? Can't we just catch `PSQLException` and fail without further checks?

Reply:
I used `if (expected.getCause() instanceof PSQLException)` and it didn't work oddly! So I have added a SQL state check. Also, note that `PSQLException` is a general DB exception and it may not necessarily be a DB connection error. Therefore, it would be better to check SQL states for more precise DB error handling.

Reply:
Ok, as it turns out, the thrown exception is a `DataAccessException` (jOOQ), not a `PSQLException`(Postgres). I really do not like this parsing of the SQL state, because as soon as a tiny detail changes, our handling will fail without us knowing. I would prefer to just catch *all* `DataAccessExceptions` and wrap them in an `UnresolvableError`.

Reply:
I have just looked a bit into the details:
1) If it should indeed be required to distinguish exception states, please do not use `sqlState()` as this is vendor specific, but use `sqlStateClass()`, which is generic. On top of that, the latter can be conveniently checked through jOOQ's `SQLStateClass` enumeration, so we do not need to have magic strings like `"57P0"` in the code.
2) That being said, all `"57P0"` cases seem to cover serious database problems (e.g., DB shutdown) and not just transaction locks. As a result, I would advise to keep the simple solution for now that I have stated before (just wrap all `DataAccessExceptions` in `UnresolvableErrors`), because it is the quickest solution that guarantees data consistency. Once it is live, we should watch the logs closely, should there be indeed a locking problem, we can always try to find a solution that is less wasteful and does not restart the pod for every mistake.

Reply:
Seems to be resolved in latest changes.